File: CodeDOM/CDOM_XplDirectoutput.cpp
/*-------------------------------------------------
 *
 *	Este archivo fue generado automáticamente.
 *	Fecha: 09/04/2009 17:21:33
 *
 *	Generado por Zoe CodeDOM Generator.
 *	COPYRIGHT 2002. por Alexis Ferreyra.
 *
 *------------------------------------------------*/

#ifndef __LAYERD_CODEDOM_ZOEDIRECTOUTPUT_V1_0_CPP
#define __LAYERD_CODEDOM_ZOEDIRECTOUTPUT_V1_0_CPP
#include "CDOM_XplDirectoutput.h"

namespace CodeDOM{

//Region de Constructores Publicos
XplDirectoutput::XplDirectoutput(){
	p_TargetPlatform = DT_EMPTY;
	p_TargetPlatformDetail = DT_EMPTY;
	p_output = DT_EMPTY;
	p_doc = DT_EMPTY;
	p_helpURL = DT_EMPTY;
	p_ldsrc = DT_EMPTY;
	p_iny = false;
	p_inydata = DT_EMPTY;
	p_inyby = DT_EMPTY;
	p_lddata = DT_EMPTY;
}
XplDirectoutput::XplDirectoutput(string n_TargetPlatform, string n_TargetPlatformDetail, string n_output){
	p_TargetPlatform = DT_EMPTY;
	p_TargetPlatformDetail = DT_EMPTY;
	p_output = DT_EMPTY;
	p_doc = DT_EMPTY;
	p_helpURL = DT_EMPTY;
	p_ldsrc = DT_EMPTY;
	p_iny = false;
	p_inydata = DT_EMPTY;
	p_inyby = DT_EMPTY;
	p_lddata = DT_EMPTY;
	set_TargetPlatform(n_TargetPlatform);
	set_TargetPlatformDetail(n_TargetPlatformDetail);
	set_output(n_output);
}
XplDirectoutput::XplDirectoutput(string n_TargetPlatform, string n_TargetPlatformDetail, string n_output, string n_doc, string n_helpURL, string n_ldsrc, bool n_iny, string n_inydata, string n_inyby, string n_lddata){
	set_TargetPlatform(n_TargetPlatform);
	set_TargetPlatformDetail(n_TargetPlatformDetail);
	set_output(n_output);
	set_doc(n_doc);
	set_helpURL(n_helpURL);
	set_ldsrc(n_ldsrc);
	set_iny(n_iny);
	set_inydata(n_inydata);
	set_inyby(n_inyby);
	set_lddata(n_lddata);
}
//Destructor
XplDirectoutput::~XplDirectoutput(){
#ifdef __LAYERD_CODEDOM_DEBUG_DESTRUCTORS
	__LAYERD_CODEDOM_DEBUG_OUTPUT("En el Destructor de 'XplDirectoutput'");
#endif
}

//Funciones Sobreescritas de XplNode
XplNode* XplDirectoutput::Clone(){
	XplDirectoutput* copy = new XplDirectoutput();
	copy->set_TargetPlatform(this->p_TargetPlatform);
	copy->set_TargetPlatformDetail(this->p_TargetPlatformDetail);
	copy->set_output(this->p_output);
	copy->set_doc(this->p_doc);
	copy->set_helpURL(this->p_helpURL);
	copy->set_ldsrc(this->p_ldsrc);
	copy->set_iny(this->p_iny);
	copy->set_inydata(this->p_inydata);
	copy->set_inyby(this->p_inyby);
	copy->set_lddata(this->p_lddata);
	copy->set_ElementName(this->get_ElementName());
	return (XplNode*)copy;
}
bool XplDirectoutput::Write(XplWriter* writer){
	bool result=true;
	//Escribo el encabezado del elemento
	writer->write(DT("<")+this->get_ElementName());
	//Escribo los atributos del elemento
	if(p_TargetPlatform != DT_EMPTY)
		writer->write(DT(" TargetPlatform=\"")+CODEDOM_Att_ToString(p_TargetPlatform)+DT("\""));
	if(p_TargetPlatformDetail != DT_EMPTY)
		writer->write(DT(" TargetPlatformDetail=\"")+CODEDOM_Att_ToString(p_TargetPlatformDetail)+DT("\""));
	if(p_output != DT_EMPTY)
		writer->write(DT(" output=\"")+CODEDOM_Att_ToString(p_output)+DT("\""));
	if(p_doc != DT_EMPTY)
		writer->write(DT(" doc=\"")+CODEDOM_Att_ToString(p_doc)+DT("\""));
	if(p_helpURL != DT_EMPTY)
		writer->write(DT(" helpURL=\"")+CODEDOM_Att_ToString(p_helpURL)+DT("\""));
	if(p_ldsrc != DT_EMPTY)
		writer->write(DT(" ldsrc=\"")+CODEDOM_Att_ToString(p_ldsrc)+DT("\""));
	if(p_iny != false)
		writer->write(DT(" iny=\"")+CODEDOM_Att_ToString(p_iny)+DT("\""));
	if(p_inydata != DT_EMPTY)
		writer->write(DT(" inydata=\"")+CODEDOM_Att_ToString(p_inydata)+DT("\""));
	if(p_inyby != DT_EMPTY)
		writer->write(DT(" inyby=\"")+CODEDOM_Att_ToString(p_inyby)+DT("\""));
	if(p_lddata != DT_EMPTY)
		writer->write(DT(" lddata=\"")+CODEDOM_Att_ToString(p_lddata)+DT("\""));
	writer->write(DT(">"));
	//Escribo recursivamente cada elemento hijo
	//Cierro el elemento
	writer->write(DT("</")+this->get_ElementName()+DT(">"));
	return result;
}
XplNode* XplDirectoutput::Read(XplReader* reader){
	throw new CodeDOM_NotImplementedException(DT("Read"));
}

//Funciones para obtener Atributos
string XplDirectoutput::get_TargetPlatform(){
	return p_TargetPlatform;
}
string XplDirectoutput::get_TargetPlatformDetail(){
	return p_TargetPlatformDetail;
}
string XplDirectoutput::get_output(){
	return p_output;
}
string XplDirectoutput::get_doc(){
	return p_doc;
}
string XplDirectoutput::get_helpURL(){
	return p_helpURL;
}
string XplDirectoutput::get_ldsrc(){
	return p_ldsrc;
}
bool XplDirectoutput::get_iny(){
	return p_iny;
}
string XplDirectoutput::get_inydata(){
	return p_inydata;
}
string XplDirectoutput::get_inyby(){
	return p_inyby;
}
string XplDirectoutput::get_lddata(){
	return p_lddata;
}

//Funciones para setear Atributos
string XplDirectoutput::set_TargetPlatform(string new_TargetPlatform){
	string back_TargetPlatform = p_TargetPlatform;
	p_TargetPlatform = new_TargetPlatform;
	return back_TargetPlatform;
}
string XplDirectoutput::set_TargetPlatformDetail(string new_TargetPlatformDetail){
	string back_TargetPlatformDetail = p_TargetPlatformDetail;
	p_TargetPlatformDetail = new_TargetPlatformDetail;
	return back_TargetPlatformDetail;
}
string XplDirectoutput::set_output(string new_output){
	string back_output = p_output;
	p_output = new_output;
	return back_output;
}
string XplDirectoutput::set_doc(string new_doc){
	string back_doc = p_doc;
	p_doc = new_doc;
	return back_doc;
}
string XplDirectoutput::set_helpURL(string new_helpURL){
	string back_helpURL = p_helpURL;
	p_helpURL = new_helpURL;
	return back_helpURL;
}
string XplDirectoutput::set_ldsrc(string new_ldsrc){
	string back_ldsrc = p_ldsrc;
	p_ldsrc = new_ldsrc;
	return back_ldsrc;
}
bool XplDirectoutput::set_iny(bool new_iny){
	bool back_iny = p_iny;
	p_iny = new_iny;
	return back_iny;
}
string XplDirectoutput::set_inydata(string new_inydata){
	string back_inydata = p_inydata;
	p_inydata = new_inydata;
	return back_inydata;
}
string XplDirectoutput::set_inyby(string new_inyby){
	string back_inyby = p_inyby;
	p_inyby = new_inyby;
	return back_inyby;
}
string XplDirectoutput::set_lddata(string new_lddata){
	string back_lddata = p_lddata;
	p_lddata = new_lddata;
	return back_lddata;
}

}

#endif

File: lib_layerd_xpl_codedom_cs/CDOM_XplDirectoutput.cs
/*-------------------------------------------------
 *
 *	Este archivo fue generado automáticamente.
 *	Fecha: 8/11/2011 4:27:19 PM
 *
 *	Generado por Zoe CodeDOM Generator para C#.
 *	COPYRIGHT 2002,2005-2006. por Alexis Ferreyra.
 *
 *------------------------------------------------*/

using System;
using System.Xml;
using System.IO;

namespace LayerD.CodeDOM{

[Serializable]
public class XplDirectoutput:  XplNode{

	#region Variables privadas para atributos y elementos
	string p_TargetPlatform;
	string p_TargetPlatformDetail;
	string p_output;
	string p_doc;
	string p_helpURL;
	string p_ldsrc;
	bool p_iny;
	string p_inydata;
	string p_inyby;
	string p_lddata;
	#endregion

	#region Region de Constructores Publicos
	public XplDirectoutput(){
		p_TargetPlatform = "";
		p_TargetPlatformDetail = "";
		p_output = "";
		p_doc = "";
		p_helpURL = "";
		p_ldsrc = "";
		p_iny = false;
		p_inydata = "";
		p_inyby = "";
		p_lddata = "";
}
	public XplDirectoutput(string n_TargetPlatform, string n_TargetPlatformDetail, string n_output){
		p_TargetPlatform = "";
		p_TargetPlatformDetail = "";
		p_output = "";
		p_doc = "";
		p_helpURL = "";
		p_ldsrc = "";
		p_iny = false;
		p_inydata = "";
		p_inyby = "";
		p_lddata = "";
		set_TargetPlatform(n_TargetPlatform);
		set_TargetPlatformDetail(n_TargetPlatformDetail);
		set_output(n_output);
	}
	public XplDirectoutput(string n_TargetPlatform, string n_TargetPlatformDetail, string n_output, string n_doc, string n_helpURL, string n_ldsrc, bool n_iny, string n_inydata, string n_inyby, string n_lddata){
		set_TargetPlatform(n_TargetPlatform);
		set_TargetPlatformDetail(n_TargetPlatformDetail);
		set_output(n_output);
		set_doc(n_doc);
		set_helpURL(n_helpURL);
		set_ldsrc(n_ldsrc);
		set_iny(n_iny);
		set_inydata(n_inydata);
		set_inyby(n_inyby);
		set_lddata(n_lddata);
	}
	#endregion

	#region Destructor
/*	public ~XplDirectoutput(){
		Se supone que .NET no necesita destructores sólo para liberar memoria	}*/
	#endregion

	#region Funciones Sobreescritas de XplNode
	public override XplNode Clone(){
		XplDirectoutput copy = new XplDirectoutput();
		copy.set_TargetPlatform(this.p_TargetPlatform);
		copy.set_TargetPlatformDetail(this.p_TargetPlatformDetail);
		copy.set_output(this.p_output);
		copy.set_doc(this.p_doc);
		copy.set_helpURL(this.p_helpURL);
		copy.set_ldsrc(this.p_ldsrc);
		copy.set_iny(this.p_iny);
		copy.set_inydata(this.p_inydata);
		copy.set_inyby(this.p_inyby);
		copy.set_lddata(this.p_lddata);
		copy.set_ElementName(this.get_ElementName());
		return (XplNode)copy;
	}

	public override CodeDOMTypes get_TypeName(){return CodeDOMTypes.XplDirectoutput;}

	public override bool Write(XplWriter writer){
		bool result=true;
		//Escribo el encabezado del elemento
		writer.WriteStartElement( this.get_ElementName() );
		//Escribo los atributos del elemento
		if(p_TargetPlatform != "")
			writer.WriteAttributeString( "TargetPlatform" ,ZoeHelper .Att_ToString(p_TargetPlatform) );
		if(p_TargetPlatformDetail != "")
			writer.WriteAttributeString( "TargetPlatformDetail" ,ZoeHelper .Att_ToString(p_TargetPlatformDetail) );
		if(p_output != "")
			writer.WriteAttributeString( "output" ,ZoeHelper .Att_ToString(p_output) );
		if(p_doc != "")
			writer.WriteAttributeString( "doc" ,ZoeHelper .Att_ToString(p_doc) );
		if(p_helpURL != "")
			writer.WriteAttributeString( "helpURL" ,ZoeHelper .Att_ToString(p_helpURL) );
		if(p_ldsrc != "")
			writer.WriteAttributeString( "ldsrc" ,ZoeHelper .Att_ToString(p_ldsrc) );
		if(p_iny != false)
			writer.WriteAttributeString( "iny" ,ZoeHelper .Att_ToString(p_iny) );
		if(p_inydata != "")
			writer.WriteAttributeString( "inydata" ,ZoeHelper .Att_ToString(p_inydata) );
		if(p_inyby != "")
			writer.WriteAttributeString( "inyby" ,ZoeHelper .Att_ToString(p_inyby) );
		if(p_lddata != "")
			writer.WriteAttributeString( "lddata" ,ZoeHelper .Att_ToString(p_lddata) );
		//Escribo recursivamente cada elemento hijo
		//Cierro el elemento
		writer.WriteEndElement();
		return result;
	}

	public override XplNode Read(XplReader reader){
		this.set_ElementName( reader.Name );
		//Lectura de Atributos
		if( reader.HasAttributes ){
			for(int i=0; i < reader.AttributeCount ; i++){
				reader.MoveToAttribute(i);
				switch(reader.Name){
					case "TargetPlatform":
						this.set_TargetPlatform(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					case "TargetPlatformDetail":
						this.set_TargetPlatformDetail(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					case "output":
						this.set_output(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					case "doc":
						this.set_doc(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					case "helpURL":
						this.set_helpURL(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					case "ldsrc":
						this.set_ldsrc(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					case "iny":
						this.set_iny(ZoeHelper .StringAtt_To_BOOL(reader.Value));
						break;
					case "inydata":
						this.set_inydata(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					case "inyby":
						this.set_inyby(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					case "lddata":
						this.set_lddata(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					default:
						throw new CodeDOM_Exception("Linea: "+reader.LineNumber+".Atributo '"+reader.Name+"' invalido en elemento '"+this.get_ElementName()+"'.");
				}
			}
			reader.MoveToElement();
		}
		//Lectura de Elementos
		if(!reader.IsEmptyElement){
		reader.Read();
		while(reader.NodeType!=XmlNodeType.EndElement){
			XplNode tempNode=null;
			switch(reader.NodeType){
				case XmlNodeType.Element:
				case XmlNodeType.Text:
					throw new CodeDOM_Exception("Linea: "+reader.LineNumber+".No se esperaba texto o elemento hijo en este nodo.");
				case XmlNodeType.EndElement:
					break;
				default:
					break;
			}
			reader.Read();
		}
		}
		return this;
	}
	public override bool BinaryWrite(XplBinaryWriter writer){
		bool result=true;
		//Escribo el ID y el nombre del elemento
		writer.Write((int) 116 );
		writer.Write( this.get_ElementName() );
		//Escribo los atributos del elemento
		writer.Write( p_TargetPlatform );
		writer.Write( p_TargetPlatformDetail );
		writer.Write( p_output );
		writer.Write( p_doc );
		writer.Write( p_helpURL );
		writer.Write( p_ldsrc );
		writer.Write( p_iny );
		writer.Write( p_inydata );
		writer.Write( p_inyby );
		writer.Write( p_lddata );
		//Escribo recursivamente cada elemento hijo
		return result;
	}

	public override XplNode BinaryRead(XplBinaryReader reader){
		this.set_ElementName( reader.ReadString() );
		//Lectura de Atributos
		p_TargetPlatform = reader.ReadString();
		p_TargetPlatformDetail = reader.ReadString();
		p_output = reader.ReadString();
		p_doc = reader.ReadString();
		p_helpURL = reader.ReadString();
		p_ldsrc = reader.ReadString();
		p_iny = reader.ReadBoolean();
		p_inydata = reader.ReadString();
		p_inyby = reader.ReadString();
		p_lddata = reader.ReadString();
		//Lectura de Elementos
		return this;
	}
/// <summary>
/// Adds arg2 node to the end of the Children collection of arg1 node.
/// </summary>
/// <param name="arg1">Node to which arg2 will be added as children.</param>
/// <param name="arg2">Node to add to arg1 as children.</param>
/// <returns>arg1 node or CodeDOM_Exception if arg1 doesn't has children.</returns>
public static XplDirectoutput operator +(XplDirectoutput arg1, XplNode arg2)
{
    if (arg1.Children() != null)
        arg1.Children().InsertAtEnd(arg2);
    else
        throw new CodeDOM_Exception("Node doesn't has children. Node can't be added.");
    return arg1;
}
/// <summary>
/// Adds arg2 nodes to the end of the Children collection of arg1 node.
/// </summary>
/// <param name="arg1">Node to which arg2 will be added as children.</param>
/// <param name="arg2">Nodes to add to arg1 as children.</param>
/// <returns>arg1 node or CodeDOM_Exception if arg1 doesn't has children.</returns>
public static XplDirectoutput operator +(XplDirectoutput arg1, XplNodeList arg2)
{
    if (arg1.Children() != null)
        arg1.Children().InsertAtEnd(arg2);
    else
        throw new CodeDOM_Exception("Node doesn't has children. Node can't be added.");
    return arg1;
}
	#endregion

	#region Funciones para obtener Atributos y Elementos
	public string get_TargetPlatform(){
		return p_TargetPlatform;
	}
	public string get_TargetPlatformDetail(){
		return p_TargetPlatformDetail;
	}
	public string get_output(){
		return p_output;
	}
	public override string get_doc(){
		return p_doc;
	}
	public override string get_helpURL(){
		return p_helpURL;
	}
	public override string get_ldsrc(){
		return p_ldsrc;
	}
	public override bool get_iny(){
		return p_iny;
	}
	public override string get_inydata(){
		return p_inydata;
	}
	public override string get_inyby(){
		return p_inyby;
	}
	public override string get_lddata(){
		return p_lddata;
	}
	#endregion

	#region Funcion ChildNodes() y Attributes()
	public override string[] Attributes(){
		string[] ret = new string[10];
		ret[0] = "TargetPlatform";
		ret[1] = "TargetPlatformDetail";
		ret[2] = "output";
		ret[3] = "doc";
		ret[4] = "helpURL";
		ret[5] = "ldsrc";
		ret[6] = "iny";
		ret[7] = "inydata";
		ret[8] = "inyby";
		ret[9] = "lddata";
		return ret;
	}
	public override string AttributeValue(string attributeName){
		if (attributeName=="TargetPlatform") return p_TargetPlatform.ToString();
		if (attributeName=="TargetPlatformDetail") return p_TargetPlatformDetail.ToString();
		if (attributeName=="output") return p_output.ToString();
		if (attributeName=="doc") return p_doc.ToString();
		if (attributeName=="helpURL") return p_helpURL.ToString();
		if (attributeName=="ldsrc") return p_ldsrc.ToString();
		if (attributeName=="iny") return p_iny.ToString();
		if (attributeName=="inydata") return p_inydata.ToString();
		if (attributeName=="inyby") return p_inyby.ToString();
		if (attributeName=="lddata") return p_lddata.ToString();
		return null;
	}
	public override XplNodeList ChildNodes(){
		XplNodeList list = new XplNodeList();
		return list;
	}
	#endregion

	#region Funciones para setear Atributos y Elementos
	public string set_TargetPlatform(string new_TargetPlatform){
		string back_TargetPlatform = p_TargetPlatform;
		p_TargetPlatform = new_TargetPlatform;
		return back_TargetPlatform;
	}
	public string set_TargetPlatformDetail(string new_TargetPlatformDetail){
		string back_TargetPlatformDetail = p_TargetPlatformDetail;
		p_TargetPlatformDetail = new_TargetPlatformDetail;
		return back_TargetPlatformDetail;
	}
	public string set_output(string new_output){
		string back_output = p_output;
		p_output = new_output;
		return back_output;
	}
	public override string set_doc(string new_doc){
		string back_doc = p_doc;
		p_doc = new_doc;
		return back_doc;
	}
	public override string set_helpURL(string new_helpURL){
		string back_helpURL = p_helpURL;
		p_helpURL = new_helpURL;
		return back_helpURL;
	}
	public override string set_ldsrc(string new_ldsrc){
		string back_ldsrc = p_ldsrc;
		p_ldsrc = new_ldsrc;
		return back_ldsrc;
	}
	public override bool set_iny(bool new_iny){
		bool back_iny = p_iny;
		p_iny = new_iny;
		return back_iny;
	}
	public override string set_inydata(string new_inydata){
		string back_inydata = p_inydata;
		p_inydata = new_inydata;
		return back_inydata;
	}
	public override string set_inyby(string new_inyby){
		string back_inyby = p_inyby;
		p_inyby = new_inyby;
		return back_inyby;
	}
	public override string set_lddata(string new_lddata){
		string back_lddata = p_lddata;
		p_lddata = new_lddata;
		return back_lddata;
	}
	#endregion

	#region Funciones para obtener nuevos nodos hijos
	#endregion

}	//Fin declaración de Clase

}

/*-------Fin de Archivo Generado------------------*/

